addons/sale/tests/test_sale_to_invoice.py

```diff
@@ -460,6 +460,121 @@ def test_invoice_analytic_account_so_not_default(self):
         aml = self.env['account.move.line'].search([('move_id', 'in', so.invoice_ids.ids)])[0]
         self.assertRecordValues(aml, [{'analytic_account_id': analytic_account_so.id}])
 
+    def test_invoice_analytic_tag_so_not_default(self):
+        """
+        Tests whether, when an analytic tag rule is set and
+        the so has an analytic tag different from default,
+        the default analytic tag doesn't get overriden in invoice.
+        """
+        self.env.user.groups_id += self.env.ref('analytic.group_analytic_accounting')
+        self.env.user.groups_id += self.env.ref('analytic.group_analytic_tags')
+        analytic_account_default = self.env['account.analytic.account'].create({'name': 'default'})
+        analytic_tag_default = self.env['account.analytic.tag'].create({'name': 'default'})
+        analytic_tag_super = self.env['account.analytic.tag'].create({'name': 'Super Tag'})
+
+        self.env['account.analytic.default'].create({
+            'analytic_id': analytic_account_default.id,
+            'analytic_tag_ids': [(6, 0, analytic_tag_default.ids)],
+            'product_id': self.product_a.id,
+        })
+        so_form = Form(self.env['sale.order'])
```

Comment:
Can't you create a SO with create instead of Form?

Reply:
Yep, I just changed it.